File: VitalSphere/VitalSphere.Services/Database/WellnessServiceCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace VitalSphere.Services.Database
{
    public class WellnessServiceCategory
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public byte[]? Image { get; set; }

        public ICollection<WellnessService> WellnessServices { get; set; } = new List<WellnessService>();
    }
}

File: VitalSphere/UI/vital_sphere_desktop/lib/model/wellness_service_category.dart
import 'package:json_annotation/json_annotation.dart';

part 'wellness_service_category.g.dart';

@JsonSerializable()
class WellnessServiceCategory {
  final int id;
  final String name;
  final String? description;
  final String? image;
  final bool isActive;
  final DateTime createdAt;

  WellnessServiceCategory({
    this.id = 0,
    this.name = '',
    this.description,
    this.image,
    this.isActive = true,
    required this.createdAt,
  });

  factory WellnessServiceCategory.fromJson(Map<String, dynamic> json) =>
      _$WellnessServiceCategoryFromJson(json);
  Map<String, dynamic> toJson() => _$WellnessServiceCategoryToJson(this);
}

